ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -1,11 +1,134 @@
+  logic, even if we have some spans of time in which a shutdown will make the
+  node start from an older point.
+
+  It is responsibility of the
+  @Ouroboros.Conensus.Storage.ChainDB.Impl.Background.copyAndSnapshotRunner@
+  copying the blocks to the disk and when dictated by the @DiskPolicy@, take a
+  snapshot. Said snapshot will /always/ be taken from the current anchor of the
+  anchored sequence, as said block is put in the ImmutableDB by the runner
+  before taking the snapshot.
+
+  The @DiskPolicy@ will specify how many snapshots to keep around, as it is
+  possible to start from an older snapshot (due to them being self contained),
+  and also how much time we want at least between snapshots, in order to avoid
+  flushing too quickly.
+
+  New-style Snapshot and flushing
+  ===============================
+
+  In the new-style database we hace mainly 3 moving pieces: the sequence of
+  @ExtLedgerState blk EmptyMK@ states, the sequence of @TableDiff@s and the disk
+  backend.
+
+  Ledger states no longer contain a full UTxO and therefore they are dependent
+  on the history before them to make a consistent view of the UTxO after
+  applying a block. The list of can no longer be pruned to @k@ as we cannot lose
+  the sequence of differences between what was last flushed and the @length -
+  k@th block.
+
+  So adapting the example above, we can have the following situation:
+
+  >                                                |<---  k blocks --->|
+  > Selected Chain:   Genesis |> ... > A >  B > C  > D > E > F > ... > T
+  > ImmutableDB:      Genesis |> ... > A >  B > C
+  > VolatileDB:                                 C |> D > E > F > ... > T  (# blocks [D,T] <= k)
+  > DiskBackend:                       *
+  > LedgerStates:                      A |> B > C >  D > E > F > ... > T
+  > LedgerDiffs:                            B > C >  D > E > F > ... > T
+
+  Note that the LedgerDiffs is a list and not an anchored sequence, and
+  therefore can be empty.
+
+  * Flushing :: applying a sequence of differences to the disk backend to make
+    them permanent. We will abuse the name and say that we "flush a block" to
+    mean that we flush the sequence of differences in the DbChangelog up to said
+    block (included).
+
+  * Snapshotting :: writing a serialization of a @ExtLedgerState blk EmptyMK@
+    from the list of ledger states.
+
+  Snapshotting and flushing shouldn't necessarily go together if the backend
+  provides a restore-point functionality. We could flush frequently and only
+  from time to time flush + create a restore point + snapshot. However, as we
+  don't currently have said functionality, we will go with the simplest
+  implementation which is just flushing + snapshotting from time to time. This
+  imposes some more restrictions, namely:
+
+  - There can only be one snapshot on the disk (or old ones can be just
+    ignored, but then why even keep them around)
```

Comment:
Maybe it'd be worth explaining why it does not make sense to keep multiple snapshots around: something along the lines of "a snapshot needs an on-disk state that corresponds to the same at which the snapshot was taken".